tests/subsys/net/openthread/rpc/server/commissioning/src/main.c

```diff
@@ -0,0 +1,236 @@
+	f(otDatasetGetActiveTlvs)
+
+extern uint64_t ot_thread_discover_cb_encoder(uint32_t callback_slot, uint32_t _rsv0,
+					      uint32_t _rsv1, uint32_t _ret,
+					      otActiveScanResult *result, void *context);
+
+static void nrf_rpc_err_handler(const struct nrf_rpc_err_report *report)
+{
+	zassert_ok(report->code);
+}
+
+static void tc_setup(void *f)
+{
+	mock_nrf_rpc_tr_expect_add(RPC_INIT_REQ, RPC_INIT_RSP);
+	zassert_ok(nrf_rpc_init(nrf_rpc_err_handler));
+	mock_nrf_rpc_tr_expect_reset();
+
+	FOREACH_FAKE(RESET_FAKE);
+	FFF_RESET_HISTORY();
+}
+
+/*
+ * Test reception of otThreadDiscover command.
+ * Test serialization of the result: OT_ERROR_NONE.
+ */
+ZTEST(ot_rpc_commissioning_server, test_otThreadDiscover)
+{
+	otThreadDiscover_fake.return_val = OT_ERROR_NONE;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(OT_ERROR_NONE), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_THREAD_DISCOVER, CBOR_UINT32(0x7fff800),
+					CBOR_UINT16(0x4321), CBOR_FALSE, CBOR_TRUE, 0,
+					CBOR_UINT32(0xcafeface)));
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(otThreadDiscover_fake.call_count, 1);
+	zassert_equal(otThreadDiscover_fake.arg1_val, 0x7fff800);
+	zassert_equal(otThreadDiscover_fake.arg2_val, 0x4321);
+	zassert_equal(otThreadDiscover_fake.arg3_val, false);
+	zassert_equal(otThreadDiscover_fake.arg4_val, true);
+	zassert_equal_ptr(otThreadDiscover_fake.arg5_val, NULL);
+	zassert_equal_ptr(otThreadDiscover_fake.arg6_val, (void *)0xcafeface);
+}
+
+/*
+ * Test reception of otDatasetIsCommissioned command.
+ * Test serialization of the result: true or false.
+ */
+ZTEST(ot_rpc_commissioning_server, test_otDatasetIsCommissioned)
+{
+	otDatasetIsCommissioned_fake.return_val = true;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(CBOR_TRUE), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED));
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(otDatasetIsCommissioned_fake.call_count, 1);
+
+	otDatasetIsCommissioned_fake.return_val = false;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(CBOR_FALSE), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED));
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(otDatasetIsCommissioned_fake.call_count, 2);
+}
+
+/*
+ * Test reception of otDatasetSetActiveTlvs command.
+ * Test serialization of the result: OT_ERROR_NONE.
+ */
+ZTEST(ot_rpc_commissioning_server, test_otDatasetSetActiveTlvs)
+{
+	const otOperationalDatasetTlvs expected_dataset = {
+		{LISTIFY(OT_OPERATIONAL_DATASET_MAX_LENGTH, LSFY_IDENTITY, (,))},
+		OT_OPERATIONAL_DATASET_MAX_LENGTH};
+
+	otDatasetSetActiveTlvs_fake.return_val = OT_ERROR_NONE;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(OT_ERROR_NONE), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_DATA_SET_ACTIVE_TLVS, TLVS));
+	mock_nrf_rpc_tr_expect_done();
+
+	zassert_equal(otDatasetSetActiveTlvs_fake.call_count, 1);
+	zassert_mem_equal(otDatasetSetActiveTlvs_fake.arg1_val, &expected_dataset,
```

Comment:
Validate the length as well?

Reply:
Length is part of dataset.

Reply:
Ah, right, when testing the maximum dataset, it will work indeed.